### PR TITLE
disable logger in tuner if it's disabled by user explicitly

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

small fix:
We should disable logger and not assign it to dummy logger if it's explicitly disabled by user inside Trainer. Dummy logger won't create any logs but we still have some other console logs that relies on `if self.logger and other_condition`, so if we assign it to DummyLogger `if self.logger` will be True which is a bit incorrect. Found this while debugging another issue.

Does it need a changelog or any test? since it's just a minor update.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rohitgr7** [comment]
> ```python
> def __bool__(self):
>     return False
> ```
> 
> for the DummyLogger? This should solve the issue and not become a breaking change. Turning the logger off temporarily can be a useful feature and should not require changing the user code where e.g. `self.logger.experiment.log()` is called.

@awaelchli, good suggestion but if we don't assign it during `.fit` why do we need to assign it to a `DummyLogger` during `.tune` if the user is explicitly setting it to `Trainer(logger=False (not default))`. Isn't the error expected if someone is using `self.logger.experiment.something`?

I don't think it will be a breaking change because if someone is setting `Trainer(logger=False)` then `self.logger.experiment.something` won't work for them on `trainer.fit`.

Also this won't work if someone is using `if self.logger is None` somewhere in their code.
```py
def __bool__(self):
    return False
```
